lib/policy_machine_storage_adapters/active_record.rb

```diff
@@ -772,6 +772,65 @@ def accessible_objects(user_or_attribute, operation, options = {})
       end
     end
 
+    # Version of accessible_objects which only returns objects that are
+    # ancestors of a specified root object or the object itself
+    def accessible_ancestor_objects(user_or_attribute, operation, root_object, options = {})
+      # Drop down to the root's stored policy element if not already there
+      root_object_pe = root_object.respond_to?(:stored_pe) ? root_object.stored_pe : root_object
```

Comment:
Two things:
- `root_object` is a pe either way, right? Since we want it to be a _stored_ pe here, I think `root_object_stored_pe` is a better name here.

- This works:
```ruby
 root_object_pe = root_object.try(:stored_pe) || root_object
```

Reply:
Also, testing what `root_object` responds to seems a roundabout way to get at what we actually want to figure out here, which is if it's of type [whatever stored_pe is]. If you'd prefer to keep the code similar to what's currently there, I'd advocate for `is_a?` over `responds_to?`.